File: Problems/No.2073_Time_Needed_to_Buy_Tickets/time_needed_to_buy_tickets.py
class Solution(object):
    def timeRequiredToBuy(self, tickets, k):
        """
        :type tickets: List[int]
        :type k: int
        :rtype: int
        """

        # parameters
        time_for_k = tickets[k]
        time = 0

        # start computing
        for i, t in enumerate(tickets):
            time += min(t, time_for_k)
            if (i == k):
                time_for_k -= 1
        
        return time

File: Problems/No.2073_Time_Needed_to_Buy_Tickets/time_needed_to_buy_tickets.cpp
class Solution {
public:
    int timeRequiredToBuy(vector<int>& tickets, int k) {
        
        // parameters
        int time_for_k = tickets[k];
        int time = 0;

        // start computing
        for (int i=0; i<tickets.size(); i++) {

            time += std::min(tickets[i], time_for_k);
            if (i == k) time_for_k--;

        }

        return time;

    }
};
